### PR TITLE
Fig 2.1 and Fig 2.2 pull request

### PR DESCRIPTION
**Pull request by 92215302**
Consider modifications of figures 2.1 and 2.2

### CODE REVIEW

File: Figure_02_01.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import numpy as np
 import matplotlib.pyplot as plt
 
-w = 0.2
+w = 0.2 #rad/s
```
Comment (alchemyst) [inline Figure_02_01.py]: Please follow pep8 comment style

File: Figure_02_01.py [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 s = w*1j
 G = 5*np.exp(-2*s)/(10*s + 1)
 
 t = np.linspace(0, 100, 200)
-GM = np.abs(G)
-PM = np.angle(G)
+GN = np.abs(G)      #Gain at w = 0.2 rad/s
```
Comment (alchemyst) [inline Figure_02_01.py]: GM and PM are understandable abbreviations for Gain and Phase margin. They are also use in the book.

File: Figure_02_02.py [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-from utilsplot import bode
-
-G = tf([5], [10, 1], deadtime=2)
+from control import bode_plot, tf, pade
+from numpy import linspace
 
+G = tf([5], [10, 1])        #Define the transfer function
+num, den = pade(2, n=10)     #Generate a 10th order pade approximation for 2s delay
+P = tf(num, den)            #General transfer function of the delay
+Gdelay = G*P                #Apply the delay to the transfer function
```
Comment (alchemyst) [inline Figure_02_02.py]: What is the benefit of replacing an exact version of the bode plot which handles dead time correctly with an approximation?

Comment (92215302) [inline Figure_02_02.py]: Agreed there is no benefit.

File: Figure_02_01.py (hunk 2 above)
Comment (92215302) [inline Figure_02_01.py]: The GM for this transfer function is 1.7 at 0.8 rad/s, and PM is 45.4 at 0.5 rad/s.  That which is calculated on line [9] is the System Gain at 0.2 rad/s (in which case SG would probably have been a better acronym).

**Comment by alchemyst** [comment]
You still need to change the comment style. Read [pep8](https://www.python.org/dev/peps/pep-0008/#comments)

File: Figure_02_01.py (hunk 1 above)
Comment (alchemyst) [inline Figure_02_01.py]: - [ ] Change comment to follow PEP8

File: Figure_02_01.py (hunk 2 above)
Comment (alchemyst) [inline Figure_02_01.py]: OK, I'm happy with the reasoning about the name change. I don't like embedding the value of a variable in a comment, though, also 

- [ ] Change comment to follow PEP8

File: Figure_02_01.py [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+PS = np.angle(G)    #Phase Shift at w = 0.2 rad/s
```
Comment (alchemyst) [inline Figure_02_01.py]: - [ ] Change comment to follow PEP8

File: Figure_02_02.py (hunk 3 above)
Comment (alchemyst) [inline Figure_02_02.py]: OK, so 
- [ ] remove this change.

**Comment by alchemyst** [comment]
I'm still happy to merge this if you make the changes

**Comment by 92215302** [comment]
Closed the pull request with the intention of updating & recommitting with new pull request. Is there a simpler way?

Regards
Jonathan

**Comment by alchemyst** [comment]
Any commits you add will be pulled in when I pull. A pull request does not carry the version of files at the time the pull request was made, but rather is just a request to pull. So you simply need to make changes and commit, and the pull request updates.

**Pull request merged**